algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -65,13 +65,19 @@
 #include <thrust/sort.h>
 #endif
 
-#if defined(KOKKOS_ENABLE_ONEDPL) && \
-    (ONEDPL_VERSION_MAJOR > 2022 ||  \
-     (ONEDPL_VERSION_MAJOR == 2022 && ONEDPL_VERSION_MINOR >= 2))
+#ifdef KOKKOS_ENABLE_ONEDPL
+#define KOKKOS_ONEDPL_VERSION                                 \
+  ONEDPL_VERSION_MAJOR * 10000 + ONEDPL_VERSION_MINOR * 100 + \
+      ONEDPL_VERSION_PATCH
+#define KOKKOS_ONEDPL_VERSION_GREATER_EQUAL(MAJOR, MINOR, PATCH) \
+  (KOKKOS_ONEDPL_VERSION >= ((MAJOR)*10000 + (MINOR)*100 + (PATCH)))
```

Comment:
This is not what I suggested in https://github.com/kokkos/kokkos/pull/7502#discussion_r1939937540

I would still slightly prefer having a named macro that more clearly expresses what we guard for and let us avoid to repeat the version (not blocking though).
Please add the `KOKKOS_IMPL_*` prefix.